substrate/frame/support/src/traits/tokens/asset_ops/common_strategies.rs

```diff
@@ -0,0 +1,391 @@
+//! This modules contains the common asset ops strategies.
+
+use super::*;
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_runtime::RuntimeDebug;
+
+/// The `CheckState` is a strategy that accepts an an `Inspect` value and the `Inner` strategy.
+///
+/// It is meant to be used when the asset state check should be performed
+/// in addition to the `Inner` strategy.
+/// The inspected state must be equal to the provided value.
+///
+/// The `CheckState` implements all potentially state-mutating strategies that the `Inner`
+/// implements.
+pub struct CheckState<Inspect: InspectStrategy, Inner = Unchecked>(pub Inspect::Value, pub Inner);
+impl<Inspect: InspectStrategy> CheckState<Inspect, Unchecked> {
+	pub fn expect(expected: Inspect::Value) -> Self {
```

Comment:
I don't really understand this name much. What about `new_unchecked`?

Reply:
I don't know :)

I find it confusing to write `CheckState::new_unchecked` (do we check or not? :D)

I imagined that the `expect` name suggests that we do check the state (and "expect" it to be equal to the supplied value) and then "just do" the desired operation.
I admit, however, that it also can be confusing.

For now, no alternative comes to mind

Reply:
Could be `new` and `new_with_inner` but usually you want the shorter name to be the one you use most of the time, and I guess that's the one that has the inner variant

Reply:
How about `new` and `with_inner`?

The `new` will be about "just checking the state before the action," while `with_inner` will be about "checking the state and then performing the action with the inner strategy."